File: app/assets/javascripts/app/services/email_sender.js
import http from './http';

export function update(data) {
  let body = new FormData();
  body.append('address', data.address || '');
  body.append('port', data.port || '');
  body.append('domain', data.domain || '');
  body.append('authentication', data.authentication || '');
  body.append('user_name', data.user_name || '');
  if (data.password) body.append('password', data.password);
  body.append('enable_starttls_auto', data.enable_starttls_auto);
  return http.put({ url: '/email_sender', body });
}

export function show() {
  return http.get({ url: '/email_sender' })
}

File: app/models/email_sender.rb
class EmailSender < ApplicationRecord

  after_save :update_system_settings

  validates_presence_of :address, :port, :domain, :authentication, :user_name, :password

  def self.singleton
    EmailSender.first_or_create
  end

  private

  def update_system_settings
    ActionMailer::Base.smtp_settings = {
        :address => self.address,
        :port => self.port,
        :domain => self.domain,
        :authentication => self.authentication,
        :user_name => self.user_name,
        :password => self.password,
        :enable_starttls_auto => self.enable_starttls_auto
    }
  end

end
